implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -77,110 +76,108 @@ public KeyLocationResolver(JWTAuthContextInfo authContextInfo) throws Unresolvab
 
     @Override
     public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContext) throws UnresolvableKeyException {
-        if (verificationKey != null) {
-            return verificationKey;
-        }
-        PublicKey key = tryAsJWK(jws);
-        if (key == null) {
-            key = tryAsPEM();
-        }
-        if (key == null) {
-            throw new UnresolvableKeyException("Failed to read location as any of JWK, JWKS, PEM. "
-                    + authContextInfo.getPublicKeyLocation());
-        }
-        return key;
-    }
-
-    private PublicKey tryAsPEM() {
-        try {
-            verificationKey = KeyUtils.decodePublicKey(content);
-        } catch (Exception e) {
-            log.debug("Failed to read location as PEM", e);
+        // httpsJwks may refresh itself even if the jwksRefreshInterval is set to 0 as it checks HTTPS Cache-Control header
+        // so the public key will have to be created per request to ensure the rotation (if any) is effective.
+        if (httpsJwks != null && authContextInfo.getTokenKeyId() == null)
```

Comment:
Can you please wrap it with the curly parentheses, I believe it is the pattern we use even for the single statement ifs. (and please push the comment above inside to make that code inside the curly brackets look bigger :-) and the comment will be better scoped)